src/storage/local.rs

```diff
@@ -0,0 +1,92 @@
+#![allow(missing_docs)]
+
+use std::fs;
+use std::str::FromStr;
+use std::vec::Vec;
+use std::path::{Path, PathBuf};
+
+use core::{LalResult, config_dir, ensure_dir_exists_fresh};
+
+
+#[derive(Serialize, Deserialize, Clone, Default)]
+pub struct LocalConfig {}
+
+use super::{Backend, Component};
+
+pub struct LocalBackend {
+    /// Local config
+    pub config: LocalConfig,
+    /// Cache directory
+    pub cache: String,
+}
+
+impl LocalBackend {
+    pub fn new(cfg: &LocalConfig, cache: &str) -> Self {
+        LocalBackend {
+            config: cfg.clone(),
+            cache: cache.into(),
+        }
+    }
+}
+
+/// Artifact backend trait for `LocalBackend`
+///
+/// This is intended to be used by the caching trait `CachedBackend`, but for
+/// specific low-level use cases, these methods can be used directly.
+impl Backend for LocalBackend {
+    fn get_versions(&self, name: &str, loc: &str) -> LalResult<Vec<u32>> {
+
+        let tar_dir = format!("cache/environments/{}/{}/", loc, name);
```

Comment:
probably needs to be a path that we store in the config rather than the cache dir.